plugins/adr-backend/src/service/router.ts

```diff
@@ -133,5 +133,72 @@ export async function createRouter(
     }
   });
 
+  router.get('/image', async (req, res) => {
+    const urlToProcess = req.query.url as string;
+    if (!urlToProcess) {
+      res.statusCode = 400;
+      res.json({ message: 'No URL provided' });
+      return;
+    }
+
+    const imageTypeMap: Record<string, string> = {
+      png: 'image/png',
+      jpg: 'image/jpeg',
+      jpeg: 'image/jpeg',
+      gif: 'image/gif',
+      svg: 'image/svg+xml',
+      webp: 'image/webp',
+    };
+
+    const imageType = urlToProcess.match(/\.([a-z0-9]+)(\?.*)?$/i);
+    if (!(imageType && imageType[1])) {
+      res.statusCode = 400;
+      res.json({ message: 'No URL to image' });
+      return;
+    }
+    if (!imageTypeMap[imageType[1].toLowerCase()]) {
+      res.statusCode = 400;
+      res.json({ message: `Image type ${imageType[1]} is not supported` });
+      return;
+    }
+    const contentType = imageTypeMap[imageType[1].toLowerCase()];
+
+    const cachedFileContent = (await cacheClient.get(urlToProcess)) as {
+      data: string;
+      etag: string;
+    };
+
+    try {
+      const fileGetResponse = await reader.readUrl(urlToProcess, {
+        etag: cachedFileContent?.etag,
+      });
+
+      const fileBuffer = await fileGetResponse.buffer();
+      const data = fileBuffer.toString();
+
+      await cacheClient.set(urlToProcess, {
+        data,
+        etag: fileGetResponse.etag,
+      });
+
+      res.setHeader('Content-Type', contentType);
+      res.send(fileBuffer);
+    } catch (error) {
+      if (cachedFileContent && error.name === NotModifiedError.name) {
+        const buffer = Buffer.from(cachedFileContent.data, 'utf-8');
+        res.setHeader('Content-Type', contentType);
+        res.send(buffer);
+        return;
+      }
+
+      const message = stringifyError(error);
+      logger.error(
+        `Unable to fetch ADRs images from ${urlToProcess}: ${message}`,
+      );
+      res.statusCode = 500;
```

Comment:
this isn't always a 500 as `readUrl()` might return a different error code, right?

Reply:
`readUrl()` can return a different error code but I'm not sure if we want to expose it to the frontend. 
it shouldn't be exposed imho...

Reply:
that's fair, but mapping all errors to 500 is a bit misleading

Reply:
When I was adding the new `/image` API endpoint I was looking at the [/file](https://github.com/backstage/backstage/blob/5462f485ee80631aafdb1cb509fba122b3ba2e14/plugins/adr-backend/src/service/router.ts#L97-L134) API endpoint. I thought that it would be the best to keep the new endpoint consistent to the already existing one especially when both endpoints are having similar functionalities.

Does it mean that we should also change the way `/file` endpoint is handling errors?